### PR TITLE
Add a mod_rewrite warning

### PR DESCRIPTION
**Pull request by juliensnz**
**Description (for Contributor and Core Developer)**

_There is a trend (as always in javascript) about giving actual hint about problem resolution in exception. THis works really really well (in particular with Redux and React and I would like to introduce it on the PIM._

**Here is a proposition to add more contextual error messages on the PIM to ease debug:**

In this particular case, the loading of the form_extension configuration, is the first load made to the REST API. Right now, on the 2.0 (and previous version), if you misconfigured your http server, your PIM was stuck at the loading stage and only a 404 error was displayed on the javascript console.

Before

Now, with this change, we display a more meaningful error explaining what is the real reason behind this error and how to solve it.

After

So this PR is not only about this particular case and also not about "we should add exception" to the javascript but more about "giving hint about problems reasons to the developer".

Tell me what do you think.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Comment by SamirBoulil** [comment]
Is there a way to not show this message in production ?

**Comment by anaelChardan** [comment]
I think, it's a good idea and yes this is definitely the current trend ;).

**Comment by juliensnz** [comment]
@SamirBoulil If this message show up it means that your pim is totally not usable. I don't think it's harmful btw as it does not give any system details

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/form/config-provider.js [hunk 1]
```diff
@@ -7,7 +7,11 @@ define(
 
         var loadConfig = function () {
             if (null === promise) {
-                promise = $.getJSON(Routing.generate('pim_enrich_form_extension_rest_index')).promise();
+                promise = $.getJSON(Routing.generate('pim_enrich_form_extension_rest_index')).fail(() => {
+                    throw Error(`It seems that your web server is not well configured as we were not able ` +
+                        `to load the frontend configuration. The most likely reason is that the mod_rewrite ` +
+                        `module is not installed/activated.`)
```
Comment (BitOne) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/form/config-provider.js]: activated => enabled

**Pull request merged**